srcCppCommon/PerftestPrinter.cxx

```diff
@@ -0,0 +1,483 @@
+                if(_showCPU)
+                    printf(", CPU %%");
+
+                printf("\n");
+            }
+            break;
+        case JSON:
+            if(_controlJsonInit){
+                _controlJsonInit = false;
+            } else {
+                printf(",\n\t\t{\n");
+            }
+            printf("\t\t\t\"length\":%d,\n",_dataLength);
+
+            if(_printIntervals && _outputFormat == JSON){
+                printf("\t\t\t\"intervals\":[\n");
+                _controlJsonIntervals = true;
+            }
+            break;
+        case LEGACY:
+            if(_headerPrinted && _printIntervals){
+                printf("\n\n********** New data length is %d\n",
+                        _dataLength);
+            }
+            break;
+    }
+}
+
+void PerftestPrinter::print_latency_interval(unsigned long latency,
+        double latency_ave,
+        double latency_std,
+        unsigned long latency_min,
+        unsigned long latency_max,
+        std::string outputCpu)
+{
+    switch (_outputFormat)
+    {
+        case CSV :
+            printf("%14d,%13lu,%9.0lf,%9.1lf,%9lu,%9lu", _dataLength,
+                    latency,
+                    latency_ave,
+                    latency_std,
+                    latency_min,
+                    latency_max);
+            if(_showCPU){
+                printf(",%6s", outputCpu.c_str());
+            }
+            printf("\n");
+            break;
+
+        case JSON :
+            if(_controlJsonIntervals)
+                _controlJsonIntervals = false;
+            else
+                printf(",");
+            printf("\n\t\t\t\t{\n"
+                    "\t\t\t\t\t\"latency\": %lu,\n"
+                    "\t\t\t\t\t\"latency_ave\": %1.2lf,\n"
+                    "\t\t\t\t\t\"latency_std\": %1.2lf,\n"
+                    "\t\t\t\t\t\"latency_min\": %lu,\n"
+                    "\t\t\t\t\t\"latency_max\": %lu",
+                    latency,
+                    latency_ave,
+                    latency_std,
+                    latency_min,
+                    latency_max
+                );
+            if(_showCPU){
+                printf(",\n\t\t\t\t\t\"cpu\": %s", outputCpu.c_str());
+            }
+            printf("\n\t\t\t\t}");
+            break;
+
+        case LEGACY :
+            printf("One way Latency: %6lu " PERFT_TIME_UNIT
+                    " Ave %6.0lf " PERFT_TIME_UNIT
+                    " Std %6.1lf " PERFT_TIME_UNIT
+                    " Min %6lu " PERFT_TIME_UNIT
+                    " Max %6lu " PERFT_TIME_UNIT,
+                    latency,
+                    latency_ave,
+                    latency_std,
+                    latency_min,
+                    latency_max
+                );
+            if(_showCPU){
+                printf(" CPU %s %%", outputCpu.c_str());
+            }
+            printf("\n");
+            break;
+    }
+}
+
+void PerftestPrinter::print_latency_summary(int total_sample_size,
```

Comment:
Use one line per parameter:

```
void PerftestPrinter::print_latency_summary(
        int total_sample_size,
        double latency_ave,
        double latency_std,
        unsigned long latency_min,
        unsigned long latency_max,
        unsigned long *_latency_history,
        unsigned long long count,
        double serializeTime,
        double deserializeTime,
        std::string outputCpu)
{
```

Reply:
Can we modify the outputCpu to be a number, instead of a string, you will see a comment explaining more about this in perftest_publisher.cxx

Reply:
ok